source/renderer/app/components/dapp/DappTransactionRequest.js

```diff
@@ -0,0 +1,254 @@
+const messages = defineMessages({
+  title: {
+    id: 'dapp.transaction.request.title',
+    defaultMessage: '!!!Transaction request',
+    description: '"title" in the dApp transaction request dialog',
+  },
+  subtitle: {
+    id: 'dapp.transaction.request.subtitle',
+    defaultMessage: '!!!triggered from {triggeredFrom}',
+    description: '"subtitle" in the dApp transaction request dialog',
+  },
+  fromWalletLabel: {
+    id: 'dapp.transaction.request.fromWalletLabel',
+    defaultMessage: '!!!From wallet',
+    description: '"fromWalletLabel" in the dApp transaction request dialog',
+  },
+  receiverLabel: {
+    id: 'dapp.transaction.request.receiver.label',
+    defaultMessage: '!!!Receiver',
+    description: '"receiver" in the dApp transaction request dialog',
+  },
+  walletsDropdownPlaceholder: {
+    id: 'dapp.transaction.request.walletsDropdown.placeholder',
+    defaultMessage: '!!!Select a wallet',
+    description:
+      '"walletsDropdownPlaceholder" in the dApp transaction request dialog',
+  },
+  addWalletLabel: {
+    id: 'dapp.transaction.request.walletsDropdown.addWalletLabel',
+    defaultMessage: '!!!Add a wallet',
+    description: '"addWalletLabel" in the dApp transaction request dialog',
+  },
+  transactionFeeLabel: {
+    id: 'dapp.transaction.request.transactionFee.label',
+    defaultMessage: '!!!Transaction fee',
+    description: '"transactionFeeLabel" in the dApp transaction request dialog',
+  },
+  additionalDataLabel: {
+    id: 'dapp.transaction.request.additionalData.label',
+    defaultMessage: '!!!Additional data',
+    description: '"additionalDataLabel" in the dApp transaction request dialog',
+  },
+  metaDataLabel: {
+    id: 'dapp.transaction.request.metaData.label',
+    defaultMessage: '!!!Meta data',
+    description: '"metaDataLabel" in the dApp transaction request dialog',
+  },
+  insufficientBalanceErrorMessage: {
+    id: 'dapp.transaction.request.error.notEnoughAda',
+    defaultMessage:
+      '!!!This wallet does not contain the minimum amount of {adaBalanceRequired} which is required for delegation to be available. Please select a wallet with <b>a minimum amount of {adaBalanceRequired}</b>.',
+    description:
+      '"Not enough ada" error in the dApp transaction request dialog',
+  },
+});
+
+type Props = {
+  adaAmount: BigNumber,
+  additionalData?: string,
+  address: string,
+  assets: Array<AssetToken>,
+  assetsAmounts: Array<BigNumber>,
+  intl: intlShape.isRequired,
+  metadata?: string,
+  onAddWallet: Function,
+  onClose: Function,
+  onSelectWallet: Function,
+  onSubmit: Function,
+  selectedWallet: ?Wallet,
+  transactionFee: BigNumber,
+  triggeredFrom: string,
+  wallets: Array<Wallet>,
+};
+
+const DappTransactionRequest = observer((props: Props) => {
+  const [isAdditionalDataVisible, toggleAdditionalData] = useState<boolean>(
+    false
+  );
+  const [isMetadataVisible, toggleMetadata] = useState<boolean>(false);
+  const getToggleLabel = (isVisible: boolean) =>
```

Comment:
Using useCallback you avoid re-defining the function everytime. Additionally its get memoized `const getToggleLabel = useCallback((isVisible: boolean) => { ... }))`